### ISSUE TITLE
Uncaught TypeError: Cannot read property 'firstChild' of undefined

### ISSUE THREAD

**Issue by zz4fap** [atom/atom#14151]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.15.0 x64
**Electron**: 1.3.13
**OS**: Ubuntu 14.04.5
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot read property 'firstChild' of undefined

```
At /opt/atom/resources/app.asar/src/line-numbers-tile-component.js:49

TypeError: Cannot read property 'firstChild' of undefined
    at DOMElementPool.freeDescendants (/app.asar/src/dom-element-pool.js:59:19)
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.setLineNumberInnerNodes (/app.asar/src/line-numbers-tile-component.js:157:27)
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateLineNumberNode (/app.asar/src/line-numbers-tile-component.js:181:14)
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateLineNumbers (/app.asar/src/line-numbers-tile-component.js:98:16)
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateSync (/app.asar/src/line-numbers-tile-component.js:78:19)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateTileNodes (/app.asar/src/tiled-component.js:55:19)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateSync (/app.asar/src/tiled-component.js:18:12)
    at GutterContainerComponent.module.exports.GutterContainerComponent.updateSync (/app.asar/src/gutter-container-component.js:79:25)
    at TextEditorComponent.module.exports.TextEditorComponent.updateSync (/app.asar/src/text-editor-component.js:235:39)
    at /app.asar/src/text-editor-component.js:323:28
    at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (/app.asar/src/view-registry.js:219:9)
    at /app.asar/src/view-registry.js:3:59
```

### Commands

```
     -1:11.9.0 core:copy (input.hidden-input)
  2x -1:04.4.0 editor:newline (input.hidden-input)
  2x -0:54.1.0 core:save (input.hidden-input)
     -0:34.6.0 editor:newline (input.hidden-input)
     -0:33.9.0 core:save (input.hidden-input)
  2x -0:30.5.0 core:move-down (input.hidden-input)
     -0:29.4.0 core:paste (input.hidden-input)
  2x -0:26.3.0 editor:newline (input.hidden-input)
     -0:22.7.0 core:cut (input.hidden-input)
     -0:21.3.0 core:paste (input.hidden-input)
  4x -0:19.7.0 core:backspace (input.hidden-input)
     -0:17.4.0 core:save (input.hidden-input)
  2x -0:15.9.0 core:backspace (input.hidden-input)
  2x -0:11.6.0 core:move-left (input.hidden-input)
     -0:09.7.0 core:backspace (input.hidden-input)
     -0:07.7.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.29.17 
atom-ctags 5.0.0 
atomic-rtags 1.3.0 
autocomplete-clang 0.10.0 
clang-format 1.28.0 
formatter-clang-format 0.1.2 
```

**Comment by rsese** [comment]
Thanks for the report!  Can you confirm a few things for us?

* what were you doing at the time of the error?
* is the error is reproducible?
* if reproducible, do you see the error in safe mode (atom --safe)?

**Comment by zz4fap** [comment]
Hi,

I was editing a file, normal work.
No, it happened only once.

Kind Regards,

Felipe

**Comment by rsese** [comment]
Thanks for confirming - I'll go ahead and close for now since it's not reproducible but if you can reproduce in the future, please let us know :v:

**Comment by jfesler** [comment]
Reproduced, but not reproducible.
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.17.2 x64
**Electron**: 1.3.15
**OS**: Mac OS X 10.12.5
**Thrown From**: Atom Core


### Stack Trace

Uncaught TypeError: Cannot read property 'firstChild' of undefined

```
At /Applications/Atom.app/Contents/Resources/app/src/dom-element-pool.js:59

TypeError: Cannot read property 'firstChild' of undefined
    at DOMElementPool.freeDescendants (/Applications/Atom.app/Contents/Resources/app/src/dom-element-pool.js:59:1)
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.setLineNumberInnerNodes (/Applications/Atom.app/Contents/Resources/app/src/line-numbers-tile-component.js:157:1)
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateLineNumberNode (/Applications/Atom.app/Contents/Resources/app/src/line-numbers-tile-component.js:181:1)
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateLineNumbers (/Applications/Atom.app/Contents/Resources/app/src/line-numbers-tile-component.js:98:1)
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateSync (/Applications/Atom.app/Contents/Resources/app/src/line-numbers-tile-component.js:78:1)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateTileNodes (/Applications/Atom.app/Contents/Resources/app/src/tiled-component.js:55:1)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateSync (/Applications/Atom.app/Contents/Resources/app/src/tiled-component.js:18:1)
    at GutterContainerComponent.module.exports.GutterContainerComponent.updateSync (/Applications/Atom.app/Contents/Resources/app/src/gutter-container-component.js:79:1)
    at TextEditorComponent.module.exports.TextEditorComponent.updateSync (/Applications/Atom.app/Contents/Resources/app/src/text-editor-component.js:237:1)
    at /Applications/Atom.app/Contents/Resources/app/src/text-editor-component.js:325:1
    at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (/Applications/Atom.app/Contents/Resources/app/src/view-registry.js:223:1)
    at /Applications/Atom.app/Contents/Resources/app/src/view-registry.js:3:1
```

### Commands

```
     -1:47.3.0 editor:newline (input.hidden-input)
     -1:46.9.0 core:move-down (input.hidden-input)
     -1:46.6.0 core:move-up (input.hidden-input)
  5x -1:46.1.0 core:move-right (input.hidden-input)
  2x -1:45 core:move-down (input.hidden-input)
  2x -1:44.4.0 core:move-up (input.hidden-input)
     -1:44 core:move-down (input.hidden-input)
     -1:43.5.0 core:move-left (input.hidden-input)
     -1:39.8.0 editor:newline (input.hidden-input)
  2x -1:39.5.0 core:cut (input.hidden-input)
  3x -1:38.4.0 core:move-up (input.hidden-input)
     -1:37.7.0 core:cut (input.hidden-input)
  2x -1:37.3.0 core:move-down (input.hidden-input)
  2x -1:35.8.0 core:move-up (input.hidden-input)
  3x -1:30.5.0 editor:newline (input.hidden-input)
     -0:21.6.0 core:copy (atom-notification.fatal.icon.icon-bug.native-key-bindings.has-detail.has-close.has-stack)
```

### Non-Core Packages

```
api-blueprint-preview 0.7.0 
atom-beautify 0.29.23 
busy-signal 1.3.0 
file-icons 2.1.2 
git-control 0.8.2 
go-debug 1.0.4 
go-plus 5.4.0 
go-signature-statusbar 1.2.0 
gotests 0.3.0 
hyperclick 0.0.40 
intentions 1.1.2 
language-api-blueprint 1.0.2 
linter 2.1.2 
linter-perl-check-syntax 0.8.1 
linter-ui-default 1.2.2 
minimap 4.27.0 
open-terminal-here 2.3.0 
perldoc 0.0.2 
perltidy 4.0.0 
pretty-json 1.6.3 
raml 0.2.0 
tester-go 2.0.2 
```

**Comment by npezza93** [comment]
This is happening to me when solving merge conflicts on atom-beta. 
```
TypeError: Cannot read property 'firstChild' of undefined
    at TextEditorComponent.measureLongestLineWidth (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:2097:112)
    at TextEditorComponent.measureContentDuringUpdateSync (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:372:16)
    at scheduler.readDocument (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:263:20)
    at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (/Applications/Atom Beta.app/Contents/Resources/app/src/view-registry.js:191:15)
    at /Applications/Atom Beta.app/Contents/Resources/app/src/view-registry.js:3:65
``` 
Any ideas?

**Comment by rsese** [comment]
If the error is reproducible for you specifically when resolving merge conflicts using the GitHub package, I'd suggest opening an issue in the package repository @npezza93:

https://github.com/atom/github

Please completely fill out the issue template with as much information as possible to reproduce, thanks!

**Comment by rickydazla** [comment]
@rsese are you some kind of savant?

**Comment by rickydazla** [comment]
`Uncaught TypeError: Cannot read property 'firstChild' of undefined`

Atom: 1.19.0-beta1
OS: mac Sierra 10.12.5 
Thrown From: Atom Core

Steps to reproduce: 

* "Merge into current branch..." using Github Desktop
* If a file exists with conflicts present, open it

```
    at TextEditorComponent.measureLongestLineWidth (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:2097:112)
    at TextEditorComponent.measureContentDuringUpdateSync (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:372:16)
    at TextEditorComponent.updateSync (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:270:18)
    at TextEditorComponent.didMouseWheel (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:1490:43)
```

**Comment by rsese** [comment]
Thanks @rickydazla - if you can reproduce consistently, can you open a new issue and fill out the issue template so we can try and repro?
